commit 8bf2240d686a44e218fb7fc2455c78b6501493ee (refs/changes/74/3674/3)
Author: Josh Wilsdon <josh@wilsdon.ca>
Date:   2018-03-19T11:44:23-07:00 (1 year, 7 months ago)
    
    OS-6782 add 'Bhyve Max Vcpus' to sysinfo
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/man/src/sysinfo.1m.md b/man/src/sysinfo.1m.md
index d4c78fe7..1a0adb9d 100644
--- a/man/src/sysinfo.1m.md
+++ b/man/src/sysinfo.1m.md
@@ -54,6 +54,20 @@ making changes.
 
     GZ only
 
+  "Bhyve Capable"
+
+    A boolean indicating whether the current hardware has the required features
+    in order to boot bhyve VMs.
+
+    GZ only
+
+  "Bhyve Max Vcpus"
+
+    A number indicating the maximum "vcpus" value that bhyve supports on this
+    platform. When no support is available the value will be 0.
+
+    GZ only
+
   "Boot Time"
 
     Timestamp (seconds since 1970-01-01 00:00:00 UTC) at which the system was
diff --git a/src/sysinfo b/src/sysinfo
index c648e182..67873b78 100644
--- a/src/sysinfo
+++ b/src/sysinfo
@@ -200,9 +200,15 @@ function get_smartos_cpu_info()
 function get_bhyve_capability()
 {
     Bhyve_Capable="false"
+    Bhyve_Max_Vcpus="0"
+
     if [[ -x /usr/lib/brand/bhyve/bhhwcompat ]]; then
         if /usr/lib/brand/bhyve/bhhwcompat; then
             Bhyve_Capable="true"
+            vcpus=$(/usr/lib/brand/bhyve/bhhwcompat -c | /usr/bin/grep "^[0-9]*$")
+            if [[ -n ${vcpus} ]]; then
+                Bhyve_Max_Vcpus=${vcpus}
+            fi
         fi
     fi
 }
@@ -666,6 +672,7 @@ CPU_Type='${CPU_Version}'
 CPU_Virtualization='${CPU_Virtualization}'
 CPU_Physical_Cores=${CPU_Count}
 Bhyve_Capable='${Bhyve_Capable}'
+Bhyve_Max_Vcpus=${Bhyve_Max_Vcpus}
 Nic_Tags=${NicTagList}
 Setup='${Setup_complete}'
 END
@@ -799,6 +806,7 @@ END
   "Setup": "${Setup_complete}",
   "VM Capable": ${VM_Capable},
   "Bhyve Capable": ${Bhyve_Capable},
+  "Bhyve Max Vcpus": ${Bhyve_Max_Vcpus},
   "CPU Type": "${CPU_Version}",
   "CPU Virtualization": "${CPU_Virtualization}",
   "CPU Physical Cores": ${CPU_Count},
